src/modules/cpu/kernel/vignette.hpp

```diff
@@ -0,0 +1,1377 @@
+                    srcPtrTempG += 16;
+                    srcPtrTempB += 16;
+                    dstPtrTemp += 48;
+                }
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    Rpp32s jLoc = vectorLoopCount - halfWidth;
+                    Rpp32f jLocComponent = jLoc * jLoc;
+                    Rpp32f gaussianValue = std::exp((iLocComponent + jLocComponent) * multiplier);
+
+                    dstPtrTemp[0] = (Rpp8u) RPPPIXELCHECK(nearbyintf((Rpp32f)*srcPtrTempR * gaussianValue));
+                    dstPtrTemp[1] = (Rpp8u) RPPPIXELCHECK(nearbyintf((Rpp32f)*srcPtrTempG * gaussianValue));
+                    dstPtrTemp[2] = (Rpp8u) RPPPIXELCHECK(nearbyintf((Rpp32f)*srcPtrTempB * gaussianValue));
+
+                    srcPtrTempR++;
+                    srcPtrTempG++;
+                    srcPtrTempB++;
+                    dstPtrTemp += 3;
+                }
+
+                srcPtrRowR += srcDescPtr->strides.hStride;
+                srcPtrRowG += srcDescPtr->strides.hStride;
+                srcPtrRowB += srcDescPtr->strides.hStride;
+                dstPtrRow += dstDescPtr->strides.hStride;
+            }
+        }
+
+        // Vignette with fused output-layout toggle (NHWC -> NHWC)
+        else if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NHWC))
+        {
+            Rpp8u *srcPtrRow, *dstPtrRow;
+            srcPtrRow = srcPtrChannel;
+            dstPtrRow = dstPtrChannel;
+
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp8u *srcPtrTemp, *dstPtrTemp;
+                srcPtrTemp = srcPtrRow;
+                dstPtrTemp = dstPtrRow;
+
+                Rpp32s iLoc = i - halfHeight;
+                Rpp32f iLocComponent = iLoc * iLoc;
+                __m256 pILocComponent = _mm256_set1_ps(iLocComponent);
+                __m256 pJLocComponent = _mm256_sub_ps(_mm256_setr_ps(0, 1, 2, 3, 4, 5, 6, 7), pHalfWidth);
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += 16)
+                {
+                    __m256 p[6];
+                    rpp_simd_load(rpp_load48_u8pkd3_to_f32pln3_avx, srcPtrTemp, p);                         // simd loads
+                    compute_vignette_48_host(p, pMultiplier, pILocComponent, pJLocComponent); // vignette adjustment
+                    rpp_simd_store(rpp_store48_f32pln3_to_u8pkd3_avx, dstPtrTemp, p);                       // simd stores
+                    srcPtrTemp += 48;
+                    dstPtrTemp += 48;
+                }
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    Rpp32s jLoc = vectorLoopCount - halfWidth;
+                    Rpp32f jLocComponent = jLoc * jLoc;
+                    Rpp32f gaussianValue = std::exp((iLocComponent + jLocComponent) * multiplier);
```

Comment:
We can break the exponent compute into 2 parts
std::exp((iLocComponent + jLocComponent) * multiplier) = std::exp((iLocComponent * multiplier) * std::exp((jLocComponent * multiplier)

And compute the std::exp((iLocComponent * multiplier) and std::exp((jLocComponent * multiplier) once before loads and store in temporary buffers

This will reduce the exponent computation from (rows * columns) to (rows + columns) times. Please check and make changes if feasible

Reply:
Done